### PR TITLE
Update help service to account for all `TypeDeclarationSyntax` (namespace+record+struct) as partial type

### PR DESCRIPTION
**Pull request by Youssef1313**
Parallel to https://github.com/dotnet/docs/pull/20798

### CODE REVIEW

File: src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs [hunk 1]
```diff
@@ -284,7 +284,8 @@ private static bool TryGetTextForContextualKeyword(SyntaxToken token, out string
                             text = "partialmethod_CSharpKeyword";
                             return true;
                         }
-                        else if (token.Parent.GetAncestorOrThis<ClassDeclarationSyntax>() != null)
+                        else if (token.Parent.GetAncestorOrThis<ClassDeclarationSyntax>() != null ||
+                            token.Parent.GetAncestorOrThis<RecordDeclarationSyntax>() != null)
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:288]: @davidwengier Can we remove this else if altogether and assume "partialtype_CSharpKeyword" if SyntaxKind.PartialKeyword is found on non-`MethodDeclarationSyntax`.

Comment (CyrusNajmabadi) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:288]: what about structs/interfaces?

Comment (Youssef1313) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:288]: @CyrusNajmabadi I should have caught this from my docs PR 😄 
I would prefer removing the else if part then, what do you think?

Comment (davidwengier) [reply src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs:288]: Seems like TypeDeclarationSyntax covers all bases

File: src/VisualStudio/CSharp/Test/F1Help/F1HelpTests.cs [hunk 2]
```diff
@@ -174,7 +174,17 @@ public async Task TestPartialType()
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.F1Help)]
-        public async Task TestPartialMethod()
+        public async Task TestRecordPartialType()
+        {
+            await Test_KeywordAsync(
+@"part[||]ial record C
```
Comment (davidwengier) [inline src/VisualStudio/CSharp/Test/F1Help/F1HelpTests.cs:180]: Would be good to have a test for a record with a primary constructor too

Comment (Youssef1313) [reply src/VisualStudio/CSharp/Test/F1Help/F1HelpTests.cs:180]: Added test.

File: src/VisualStudio/CSharp/Test/F1Help/F1HelpTests.cs [hunk 3]
```diff
@@ -174,7 +174,28 @@ public async Task TestPartialType()
...
+{
+    partial void goo();
+}", "partialtype");
+        }
+
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Test/F1Help/F1HelpTests.cs:185]:
```suggestion
```

**Review by davidwengier — APPROVED** [review]

**Comment by davidwengier** [comment]
Thanks!

**Pull request merged**